src/sensors/rev_encoder.cpp

```diff
@@ -1,18 +1,29 @@
-#include "rev_encoder.hpp"
+#include <rev_encoder.hpp>
 
-RevEnc::RevEnc(uint8_t pin) {
-    this->pin = pin;
-	freq.begin(this->inPin, FREQMEASUREMULTI_MARK_ONLY);
+RevEncoder::RevEncoder(uint8_t encPin, int baudrate) 
+{
+  this->inPin = encPin;
+  pinMode(this->inPin, INPUT);  // Set the pin used to measure the encoder to be an input
+  Serial.begin(baudrate);
+  freq.begin(this->inPin, FREQMEASUREMULTI_MARK_ONLY);
 }
 
-int RevEnc::getAngleRaw() {
-	while (this->freq.available() > 1) this->freq.read(); // Burn through buffer of values in freq
-	int angle = round(freq.countToNanoseconds(freq.read()) / 1000.0);
-	if (angle >= 1 && angle <= 1025) return angle;
-	return -1;
+void RevEncoder::read() 
+{
+  if (freq.available()) 
+    {
+      float frequency = this->freq.countToNanoseconds(this->freq.read()) / 1000;
+      this->ticks = (int) frequency % 1024;
```

Comment:
Here you've done the int cast I was talking about. Can you convert this into a normalized float instead so that this doesn't reduce our resolution?

Reply:
Just checked the docs you linked, the sensor has a resolution of:

"Quadrature Resolution: 2048 Cycles per Revolution (8192 Counts per Revolution)"

if you switch your 1024 to 8192 we should maintain all the resolution (make sure your frequency aligns with the new 1 revolution as well)

Reply:
Or maybe I don't understand the sensor: it's 8192 incremental outputs, but 1024 absolute frequency outputs? I'll probably leave it to you then. Either way you should be good to have ticks as an int.